scripts/build/config.mjs

```diff
@@ -16,6 +22,62 @@ const {
 } = createEsmUtils(import.meta);
 const resolveEsmModulePath = async (specifier) =>
   url.fileURLToPath(await importMetaResolve(specifier));
+const copyFileBuilder = ({ file }) =>
+  copyFile(
+    path.join(PROJECT_ROOT, file.input),
+    path.join(DIST_DIR, file.output.file)
+  );
+
+async function typesFileBuilder({ file }) {
+  /**
+   * @typedef {{ from: string, to: string }} ImportPathReplacement
+   * @typedef {{ [input: string]: Array<ImportPathReplacement> }} ReplacementMap
+   */
+
+  /** @type {Array<ImportPathReplacement>} */
+  const jsParsersImportReplacement = [
+    { from: "../../index.js", to: "../index.js" },
+  ];
+  /** @type {ReplacementMap} */
+  const pathReplacementMap = {
+    "src/document/index.d.ts": [{ from: "../index.js", to: "./index.js" }],
+    "src/language-js/parse/acorn-and-espree.d.ts": jsParsersImportReplacement,
+    "src/language-js/parse/angular.d.ts": jsParsersImportReplacement,
+    "src/language-js/parse/babel.d.ts": jsParsersImportReplacement,
+    "src/language-js/parse/flow.d.ts": jsParsersImportReplacement,
+    "src/language-js/parse/meriyah.d.ts": jsParsersImportReplacement,
+    "src/language-js/parse/typescript.d.ts": jsParsersImportReplacement,
+  };
+  const replacements = pathReplacementMap[file.input];
+  if (!replacements || replacements.length === 0) {
+    await copyFileBuilder({ file });
+    return;
+  }
+  let data = await fs.promises.readFile(file.input, "utf8");
+  for (const { from, to } of replacements) {
+    data = data.replaceAll(
+      new RegExp(` from "${from}";`, "g"),
+      ` from "${to}";`
+    );
+  }
+  await writeFile(path.join(DIST_DIR, file.output.file), data);
+}
```

Comment:
@fisker I've added `typesFileBuilder`. It runs following steps:
- Checks if a target file includes the import path should be replaced to other path.
- If no, just copy file.
- If yes, replace import file path by `pathReplacementMap` and RegExp.

Reply:
Actually we can calculate pathes (there is `files` in first argument can use to find where the entry build to), but we can improve this part later.